spec/index.bs

```diff
@@ -3487,6 +3488,7 @@ interface mixin GPUProgrammablePassEncoder {
 
     void beginPipelineStatisticsQuery(GPUQuerySet querySet, GPUSize32 queryIndex);
     void endPipelineStatisticsQuery(GPUQuerySet querySet, GPUSize32 queryIndex);
+    void writeTimestamp(GPUQuerySet querySet, GPUSize32 queryIndex);
```

Comment:
Maybe we could have it on the `GPUCommandEncoder` too so that we can time copy operations or whole render passes.

Reply:
If so, I think we can move `writeTimestamp` to `GPUCommandEncoder`, and also call `encoder.writeTimestamp` inside render pass or compute. For Metal backend, the [sampleCountersInBuffer](https://developer.apple.com/search/?q=sampleCountersInBuffer) are on blit/render/compute encoders, so we just need to know which encoder is activing in command encoder currently, then call the API of the encoder.

Reply:
I don't think that'd work because the encoder is "borrowed" by the pass and it shouldn't be possible to used it until the pass is ended. However we could maybe have a GPUEncoderBase that gets `writeTimestamp` `push/popDebugGroup` and `insertMarker` that's exposed to all encoder types.